pkg/util/utils.go

```diff
@@ -291,3 +293,31 @@ func ContainsString(slice []string, s string) bool {
 	}
 	return false
 }
+
+// ValidateUser validates minikube is run by the recommended user (privileged or regular)
+func ValidateUser(driverName string) {
+	u, err := user.Current()
+	// Check if minikube needs to run with sudo or not.
+	if err == nil {
+		if driverName == constants.DriverNone && u.Name != "root" {
```

Comment:
Instead of checking for the username `root`, check that `u.Uid != 0` here an elsewhere. The root username can be renamed, and there can be multiple users with uid 0, and thus root privileges.

Reply:
I was not aware that multiple users can have the same `User ID`. I always thought that `gid` can be same. I guess this is one of the special cases. Thanks for the info!